pepper-apis/src/main/java/org/broadinstitute/ddp/filter/StudyLanguageResolutionFilter.java

```diff
@@ -0,0 +1,120 @@
+
+    private static Map<Locale, LanguageDto> localeToLanguage = new HashMap<>();
+
+    public static final String USER_LANGUAGE = "USER_LANGUAGE";
+    private static final Logger LOG = LoggerFactory.getLogger(StudyLanguageResolutionFilter.class);
+    private static final String STUDY_GUID_REGEX = "/studies/(\\w+)";
+    private static final int STUDY_GUID_INDEX = 1;
+
+    @Override
+    public void handle(Request request, Response response) {
+        try {
+            String acceptLanguageHeader = request.headers(RouteConstants.ACCEPT_LANGUAGE);
+            Matcher matcher = Pattern.compile(STUDY_GUID_REGEX).matcher(request.url());
+            String studyGuid = matcher.find() ? matcher.group(STUDY_GUID_INDEX) : null;
+            // The "supported languages" notion is an attribute of a study, thus is doesn't
+            // make any sense outside of the study context, so we issue a warning
+            boolean supportedLanguagesCanBeDetected = studyGuid != null;
+            if (!supportedLanguagesCanBeDetected) {
+                LOG.warn(
+                        "Supported languages can't be detected because the filter is invoked"
+                        + " before the route that is outside of the study context. Please"
+                        + " remount the filter under '/user/*/studies/*' instead. Current path = {}",
+                        request.url()
+                );
+                return;
+            }
+            LOG.info("The supported languages can be detected for the study {}", studyGuid);
+            Locale ddpAuthPreferredLocale = RouteUtil.getDDPAuth(request).getPreferredLocale();
+            TransactionWrapper.useTxn(
+                    handle -> {
+                        Locale preferredLocale = StudyLanguageResolutionFilter.getPreferredLocale(
+                                handle, acceptLanguageHeader, ddpAuthPreferredLocale, studyGuid
+                        );
+                        LanguageDto preferredLanguage = StudyLanguageResolutionFilter.convertLocaleToLanguageDto(
+                                handle, preferredLocale
+                        );
+                        request.attribute(USER_LANGUAGE, preferredLanguage);
+                        LOG.info("Added the preferred user language {} to the attribute store", preferredLanguage.getIsoCode());
+                        String contentLanguageHeader = StudyLanguageResolutionFilter.createContentLanguageHeaderFromLocale(preferredLocale);
+                        response.header(RouteConstants.CONTENT_LANGUAGE, contentLanguageHeader);
+                    }
+            );
+        } catch (Exception e) {
+            LOG.error("Error while figuring out the user language", e);
+        }
+    }
+
+    static String createContentLanguageHeaderFromLocale(Locale locale) {
+        // E.g. "en-US"
+        return String.format("%s-%s", locale.getLanguage(), locale.getCountry());
```

Comment:
The languages we have in database are not actual locales (e.g. `en` and not `en-US`), so if the `Locale` is from their user profile, then I think the Locale is just `en`, so won't the content-language header become `en-`?

I think it's simpler and less error-prone to just call [`Locale.toLanguageTag()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Locale.html#toLanguageTag()).

Reply:
Done.